### PR TITLE
[DX-1647] 5.3.5 Gateway release notes

### PR DESCRIPTION
**Pull request by Eopayemi**
### For internal users - Please add a Jira DX PR ticket to the subject!

[DX-1647](https://tyktech.atlassian.net/browse/DX-1647)

---

### Preview Link

[Preview](https://deploy-preview-5329--tyk-docs.netlify.app/docs/product-stack/tyk-gateway/release-notes/version-5.3)

### Description

Gateway 5.3.5 release notes

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labeled** the PR

[DX-1647]: https://tyktech.atlassian.net/browse/DX-1647?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

___

### **PR Type**
Documentation

___

### **Description**
- Added detailed release notes for Tyk Gateway version 5.3.5, including new features and fixes.
- Introduced new configuration options for controlling URL path matching strictness to enhance request handling.
- Provided information on FIPS 140-2 compliance and updated the compatibility matrix for Tyk components and third-party dependencies.
- Documented fixes for issues related to path-based permissions and endpoint URL handling to improve authorization accuracy.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>version-5.3.md</strong><dd><code>Update Tyk Gateway 5.3.5 Release Notes with New Features and Fixes</code></dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md

<li>Added release notes for Tyk Gateway version 5.3.5.<br> <li> Introduced new configuration options for URL path matching strictness.<br> <li> Provided details on FIPS compliance and updated compatibility matrix.<br> <li> Documented fixes for path-based permissions and endpoint handling.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5329/files#diff-ead598197a92e18014f4f7c35c341621948fb6cdf66d45c1dd05de21bc2b35aa">+141/-4</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Review by dcs3spp — COMMENTED** [review]
Thanks for creating Release notes for 5.3.5 @Eopayemi. Let me know once changelog has been added and I will review

**Review by andyo-tyk — CHANGES_REQUESTED** [review]
Added change log, promoted Release Highlights section for visibility. Just needs release date to be added.

**Review by carolwkc — APPROVED** [review]
Helm chart version to be updated when it is released.

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 1]
```diff
@@ -20,29 +20,161 @@ tags: ["Tyk Gateway", "Release notes", "v5.3", "5.3.0", "5.3.1", "5.3.3", "chang
...
+
+We have introduced two new options in the `http_server_options` [Gateway configuration]({{< ref "tyk-oss-gateway/configuration#http_server_options" >}}) that will enforce prefix and/or suffix matching when Tyk performs checks on whether middleware or other logic should be applied to a request:
+
+- `enable_path_prefix_matching` ensures that the start of the request path must match the path defined in the API definition
+- `enable_path_suffix_matching` ensures that the end of the request path must match the path defined in the API definition
+- combining `enable_path_prefix_matching` and `enable_path_suffix_matching` will ensure an exact (explicit) match is performed
+
+These configuration options provide control to avoid unintended matching of paths from Tyk's default *wildcard* match. Use of regex special characters when declaring the endpoint path in the API definition will automatically override these settings for that endpoint.
+Tyk recommends that exact matching is employed, but both options default to `false` to avoid introducing a breaking change for existing users.
+</details>
```
Comment (andyo-tyk) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md:128]:
```suggestion

The example Gateway configuration file `tyk.conf.example` has been updated to set exact matching with:

- `http_server_options.enable_path_prefix_matching = true`
- `http_server_options.enable_path_suffix_matching = true`
- `http_server_options.enable_strict_routes = true`
</details>
```

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 2]
```diff
@@ -20,29 +20,167 @@ tags: ["Tyk Gateway", "Release notes", "v5.3", "5.3.0", "5.3.1", "5.3.3", "chang
...
+We have introduced two new options in the `http_server_options` [Gateway configuration]({{< ref "tyk-oss-gateway/configuration#http_server_options" >}}) that will enforce prefix and/or suffix matching when Tyk performs checks on whether middleware or other logic should be applied to a request:
+
+- `enable_path_prefix_matching` ensures that the start of the request path must match the path defined in the API definition
+- `enable_path_suffix_matching` ensures that the end of the request path must match the path defined in the API definition
+- combining `enable_path_prefix_matching` and `enable_path_suffix_matching` will ensure an exact (explicit) match is performed
+
+These configuration options provide control to avoid unintended matching of paths from Tyk's default *wildcard* match. Use of regex special characters when declaring the endpoint path in the API definition will automatically override these settings for that endpoint.
+Tyk recommends that exact matching is employed, but both options default to `false` to avoid introducing a breaking change for existing users.
+
+The example Gateway configuration file `tyk.conf.example` has been updated to set exact matching with:
```
Comment (andyo-tyk) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]:
```suggestion
The example Gateway configuration file `tyk.conf.example` has been updated to set the recommended exact matching with:
```

**Review by andyo-tyk — APPROVED** [review]

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 3]
```diff
@@ -20,29 +20,167 @@ tags: ["Tyk Gateway", "Release notes", "v5.3", "5.3.0", "5.3.1", "5.3.3", "chang
...
+
+### Release Highlights
+
+This release fixes some issues related to the way that Tyk performs URL path matching, introducing two new Gateway configuration options to control path matching strictness. For a comprehensive list of changes, please refer to the detailed [changelog]({{< ref "#Changelog-v5.3.5">}}) below.
 
 
 ### Breaking Changes
-**Attention**: Please read this section carefully.
 
+**Attention**: Please read this section carefully.
```
Comment (letzya) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]:
```suggestion
```

**Review by andyo-tyk — APPROVED** [review]

**Review by sharadregoti — APPROVED** [review]

**Review by sharadregoti — APPROVED** [review]

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 4]
```diff
@@ -20,29 +20,161 @@ tags: ["Tyk Gateway", "Release notes", "v5.3", "5.3.0", "5.3.1", "5.3.3", "chang
...
 ---
-## 5.3.4 Release Notes
 
-## Release Date August 26th 2024
+## 5.3.5 Release Notes
 
 
-### Breaking Changes
-**Attention**: Please read this section carefully.
+### Release Date 24 September 2024
```
Comment (sharadregoti) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]:
```suggestion
### Release Date 26 September 2024
```

**Review by sharadregoti — APPROVED** [review]

**Pull request merged**